chain/sub/incoming.go

```diff
@@ -565,36 +566,60 @@ func (mv *MessageValidator) Validate(ctx context.Context, pid peer.ID, msg *pubs
 }
 
 func (mv *MessageValidator) validateLocalMessage(ctx context.Context, msg *pubsub.Message) pubsub.ValidationResult {
+	ctx, _ = tag.New(
+		ctx,
+		tag.Upsert(metrics.Local, "true"),
+	)
 	// do some lightweight validation
 	stats.Record(ctx, metrics.MessagePublished.M(1))
 
 	m, err := types.DecodeSignedMessage(msg.Message.GetData())
 	if err != nil {
 		log.Warnf("failed to decode local message: %s", err)
+		ctx, _ = tag.New(
+			ctx,
+			tag.Upsert(metrics.FailureType, "decode"),
+		)
 		stats.Record(ctx, metrics.MessageValidationFailure.M(1))
```

Comment:
let's add a local function that does this combo so that we don't have too much duplication; we can call it `recordFailure`.